src/foam/dao/SQL.js

```diff
@@ -84,12 +84,17 @@ foam.CLASS({
 });
 
 
+/**
+   The base models Unary and Binary are used to validate the arguments
```

Comment:
I'm personally not wild about this setup.

This means that when Eq goes to generate its SQL, it will call SUPER() which calls arg1.toSQL() and arg2.toSQL(), then it goes to call SQLSupport.values() which also calls arg1.toSQL() and arg2.toSQL().  We're generating twice as much SQL as we need not to mention checking for the existence of arg1, arg2, arg1.toSQL, arg1.toString, arg2.toSQL, arg2.toString, multiple times.

In what cases would arg1 or arg2 not have a toSQL() method?

If we refine toSQL() on all things that can be in an expression (Properties and all mlangs) then we can reliably call toSQL() on all arguments.

arg1 and arg2 already have validation to ensure you don't set them to invalid values.